blocks/library/reusable-block/index.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * External dependencies
+ */
+import { pickBy, noop } from 'lodash';
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { Placeholder, Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getBlockType, registerBlockType } from '../../api';
+import ReusableBlockEditPanel from './edit-panel';
+
+class ReusableBlockEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.toggleEditing = this.toggleEditing.bind( this );
+		this.setAttributes = this.setAttributes.bind( this );
+		this.setName = this.setName.bind( this );
+		this.updateReusableBlock = this.updateReusableBlock.bind( this );
+
+		this.state = {
+			isEditing: false,
+			name: null,
+			attributes: null,
+		};
+	}
+
+	componentDidMount() {
+		if ( ! this.props.reusableBlock ) {
+			this.props.fetchReusableBlock();
+		}
+	}
+
+	toggleEditing() {
+		this.setState( ( prevState ) => ( {
+			isEditing: ! prevState.isEditing,
+			name: null,
+			attributes: null,
+		} ) );
+	}
+
+	setAttributes( attributes ) {
+		this.setState( ( prevState ) => ( {
+			attributes: { ...prevState.attributes, ...attributes },
+		} ) );
+	}
+
+	setName( name ) {
+		this.setState( { name } );
+	}
+
+	updateReusableBlock() {
+		const { name, attributes } = this.state;
+		// TODO: Can these two actions be combined?
+		// TODO: Think about a loading indicator, success message, failure message, etc.
+		this.props.updateReusableBlock( pickBy( { name, attributes } ) );
+		this.props.saveReusableBlock();
```

Comment:
Yes, maybe the `saveReusableBlock`action could take the updated name and attributes as arguments and optimistically update the state (similarly to how we do this for the `savePost` effect).

Reply:
The only problem is that `saveReusableBlock` is also dispatched [when converting a static block into a reusable block](https://github.com/WordPress/gutenberg/pull/2659/files#diff-c0e459b150c84795a6a30e7104d1027fR386). But maybe I could combine `addReusableBlock`, `updateReusableBlock`, **and** `saveReusableBlock` into one action... 🤔